src/main/java/frc/robot/hardware/phoenix6/Phoenix6Device.java

```diff
@@ -31,17 +31,30 @@ public boolean isConnected() {
 		return connectedInput.connected;
 	}
 
+	private InputSignal<?>[] getValidSignals(InputSignal<?>... signals) {
+		LinkedList<InputSignal<?>> signalsSet = new LinkedList<>();
+		for (InputSignal<?> signal : signals) {
+			if (signal instanceof Phoenix6SignalBuilder.SignalGetter) {
+				signalsSet.add(signal);
+			} else {
+				new Alert(
+					Alert.AlertType.WARNING,
+					logPath + "signal named " + signal.getName() + " got invalid type " + signal.getClass().getSimpleName()
+				).report();
+			}
+		}
+		return signalsSet.toArray(new InputSignal<?>[0]);
+	}
+
+
 	private StatusCode refreshSignals(InputSignal<?>... signals) {
 		LinkedList<StatusSignal<Double>> signalsSet = new LinkedList<>();
-		for (int i = 0; i < signals.length; i++) {
-			if (signals[i] instanceof Phoenix6SignalBuilder.SignalGetter signalGetter) {
+		for (InputSignal<?> signal : signals) {
+			if (signal instanceof Phoenix6SignalBuilder.SignalGetter signalGetter) {
 				signalsSet.add(signalGetter.getSignal());
-				if (signals[i] instanceof Phoenix6BothLatencySignal bothLatencySignal) {
+				if (signal instanceof Phoenix6BothLatencySignal bothLatencySignal) {
 					signalsSet.add(bothLatencySignal.getSignalSlope());
 				}
```

Comment:
why do you need the if's if you already check that in the valid signals? and shouldnt the if of the both latency should be in the validsignals as well?

Reply:
No reason to put the if there, I only check if it’s a phoenix signal.
About the if It is not a must u can just cast but I like it with the check, like that the function will work by itself without the need to run valid before

Reply:
yeah ok i dont really see why do you want this to work by itself if its a private function that its only use is in the update signals but its your choice if you think its needed, i wont get stuck on it